rosdep/python.yaml

```
@@ -9382,7 +9383,9 @@ python3-uvloop:
   ubuntu: [python3-uvloop]
 python3-vcstool:
   alpine: [vcstool]
-  debian: [python3-vcstool]
+  debian:
+    bookworm: [vcstool]
+    '*': [python3-vcstool]
```

These entries need to be alphabetized as well.
```suggestion
    '*': [python3-vcstool]
    bookworm: [vcstool]
```

---

Actually, on second thought, I'm not sure this is what we want to do. The `vcstool` package is provided by debian upstream, but we package it as part of the ROS repositories under the name `python3-vcstool`. I'm pretty sure that's what we're intending to install here.

@nuclearsandwich to confirm.

---

+1 the rosdep database should continue to prefer the packages from the ROS bootstrap repository (available via the ROS and ROS 2 repositories). @sprhawk if you prefer the upstream vcstool you can use custom rosdep sources to override the upstream database.

---

I don't understand clearly. I thought these names are used in specified distribution (vcstool in debian bookworm), not used (python3-vcstool) in ROS1/2 repo, even the package name used in alpine is also `vcstool`, why shouldn't it be changed for bookworm ?

---

Don't understand. I think the list here is for installing packages via system package manager, but bookworm doesn't ship `python3-vcstool`, but `vcstool`, why it is related to ROS / ROS2 repositories ?

---

When running rosdep, we assume that all of the package sources listed under a platform's expected sources in the rosdep documentation are available. The [acceptable sources for packages listed for Debian](https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#debian) include ROS-provided repositories. In addition to providing deb packages for actual ROS packages, these repositories provide up-to-date versions of ROS infrastructure packages (i.e. rosdep, catkin_pkg, rosinstall_generator, bloom, catkin-tools, colcon, etc).

Even when packages are available "upstream", we "prefer" the packages provided by the ROS repositories. Debian decided to name their "upstream" package differently from ours, but the rule is nonetheless intended to target the ROS-provided package and not the Debian-provided one.

---

This still needs to be fixed to `python3-vcstool`.